From 440a27eeca886cd2d0c97f77a67316887a07c011 Mon Sep 17 00:00:00 2001
From: Ajay Kaher <akaher@vmware.com>
Date: Wed, 15 Mar 2023 17:34:24 +0000
Subject: [PATCH] Revert "board_r: Relocate OF_EMBED if NEEDS_MANUAL_RELOC
 only"

This reverts commit 76f921eb95d5b814f973a263187db509d6f03903.

RPI3 board fail to boot with 76f921eb95d5b814f973a263187db509d6f03903,
so reverting commit 76f921eb95d5b814f973a263187db509d6f03903.
---
 common/board_r.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/common/board_r.c b/common/board_r.c
index 00926dcb1e..56eb60fa27 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -150,13 +150,13 @@ static int initr_reloc_global_data(void)
 	 */
 	gd->env_addr += gd->reloc_off;
 #endif
+#ifdef CONFIG_OF_EMBED
 	/*
 	 * The fdt_blob needs to be moved to new relocation address
 	 * incase of FDT blob is embedded with in image
 	 */
-	if (CONFIG_IS_ENABLED(OF_EMBED) && CONFIG_IS_ENABLED(NEEDS_MANUAL_RELOC))
-		gd->fdt_blob += gd->reloc_off;
-
+	gd->fdt_blob += gd->reloc_off;
+#endif
 #ifdef CONFIG_EFI_LOADER
 	/*
 	 * On the ARM architecture gd is mapped to a fixed register (r9 or x18).
-- 
2.19.0

